src/Compilers/VisualBasic/BasicAnalyzerDriver/VisualBasicDeclarationComputer.vb

```diff
@@ -172,10 +172,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return initializer
         End Function
 
-        Private Shared Function GetAsNewClauseIntializer(asClause As AsClauseSyntax) As ExpressionSyntax
-            Return If(asClause.IsKind(SyntaxKind.AsNewClause),
-                CType(asClause, AsNewClauseSyntax).NewExpression,
-                Nothing)
+        Private Shared Function GetAsNewClauseIntializer(asClause As AsClauseSyntax) As SyntaxNode
+            ' The As New clause itself is necessary rather than the embedded New expression, so that the
+            ' code block associated with the declaration appears as an initializer for the purposes
+            ' of executing analyzer actions.
+            Return If(asClause.IsKind(SyntaxKind.AsNewClause), asClause, Nothing)
```

Comment:
This edit causes As New declarations to be treated as having member initializers by semantic models (rather than as just having object creation expressions). This could be patched within the semantic model code instead.